src/backend/distributed/transaction/transaction_recovery.c

```diff
@@ -171,9 +172,12 @@ RecoverWorkerTransactions(WorkerNode *workerNode)
 
 	MemoryContext oldContext = MemoryContextSwitchTo(localContext);
 
-	/* take table lock first to avoid running concurrently */
-	Relation pgDistTransaction = table_open(DistTransactionRelationId(),
-											ShareUpdateExclusiveLock);
+	Oid distTransactionRelationId = DistTransactionRelationId();
+
+	/* take advisory lock first to avoid running concurrently */
+	LockShardResource((uint64) distTransactionRelationId, ShareUpdateExclusiveLock);
```

Comment:
Using this method here might not be the best option as `pg_dist_tranasction` is not a shard